### PR TITLE
DKP-539 Added release notes for Docker Desktop 4.16

### PR DESCRIPTION
**Pull request by mat007**
### Proposed changes

Added release notes for Docker Desktop 4.16

### CODE REVIEW

File: desktop/release-notes.md [hunk 1]
```diff
@@ -24,13 +24,92 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Support downloading Kubernetes images via an HTTP proxy.
+
+#### For Mac
+
+- Fixed the Docker engine not starting when Enhanced Container Isolation is enabled if the legacy `osxfs` implementation is used for file sharing.
+- Fixed files created on VirtioFS having the executable bit set. Fixes [docker/for-mac#6614](https://github.com/docker/for-win/issues/6614).
+- Added back a way to uninstall Docker Desktop from the command line. Fixes [docker/for-mac#6598](https://github.com/docker/for-mac/issues/6598).
+- Fixed hardcoded `/usr/bin/kill`. Fixes [docker/for-mac#6589](https://github.com/docker/for-mac/issues/6589).
+- Fixed truncation (for example with the `truncate` command) of very large files (> 38GB) shared on VirtioFS with an incorrect size.
+
```
Comment (djs55) [inline desktop/release-notes.md:71]:
```suggestion
- Fixed Docker crash under network load. Fixes [docker/for-mac#6530](https://github.com/docker/for-mac/issues/6530).

```

Comment (djs55) [inline desktop/release-notes.md:71]:
```suggestion
- Fixed an issue causing Docker to prompt the user to install the `/var/run/docker.sock` symlink after every reboot.

```
WDYT @aiordache ?

File: desktop/release-notes.md [hunk 2]
```diff
@@ -24,13 +24,92 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+### For Windows
+
+- Fixed `docker build` hanging while printing "load metadata for". Fixes [docker/for-win#10247](https://github.com/docker/for-win/issues/10247).
+- Fixed typo in diagnose.exe output (https://github.com/docker/for-win/issues/13107).
+- Added support for running under cgroupv2 on WSL 2. This is activated by adding `kernelCommandLine = systemd.unified_cgroup_hierarchy=1 cgroup_no_v1=all` to your `%USERPROFILE%\.wslconfig` file in the `[wsl2]` section.
+
```
Comment (djs55) [inline desktop/release-notes.md:77]:
```suggestion
- When running in WSL 2 mode, the privileged Docker Desktop Service is not required.

```
WDYT @elapuyade ?

File: desktop/release-notes.md (hunk 1 above)
Comment (mat007) [reply desktop/release-notes.md:71]: What’s the pr number for this one? Did I miss it somehow?

Comment (mat007) [reply desktop/release-notes.md:71]: Same here, is there a release note in the PR?

File: desktop/release-notes.md [hunk 3]
```diff
@@ -24,13 +24,92 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Extensions have moved from Beta to GA
+- Quick Search has moved from experimental to GA
+- Extensions are now included in Quick Search
+- Analyzing large images is now up to 4x faster
+
+### Upgrades
+
+- [Compose v2.15.0](https://github.com/docker/compose/releases/tag/v2.15.0)
+- [Containeld v1.6.14](https://github.com/containerd/containerd/releases/tag/v1.6.14)
```
Comment (trungutt) [inline desktop/release-notes.md:45]: Typo `Containerd` instead of `Containeld`

File: desktop/release-notes.md [hunk 4]
```diff
@@ -24,13 +24,92 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- [Docker Engine v20.10.22](https://docs.docker.com/engine/release-notes/#201022)
+- [Docker Scan v0.23.0](https://github.com/docker/scan-cli-plugin/releases/tag/v0.23.0)
+- [Go 1.19.4](https://github.com/golang/go/releases/tag/go1.19.4)
+
+### Bug fixes and enhancements
+
+#### For all platforms
```
Comment (trungutt) [inline desktop/release-notes.md:52]: This PR fixes an important issue https://github.com/docker/pinata/pull/19991 on the Container overview page. 
There's no corresponding Github issue but it's been reported several times on feedback-channel Slack and also [by our client](https://docker.slack.com/archives/C0W4XJVFW/p1672985651503269). I'd put it here, something like
```
- Fixed "An error occured" with Container overview screen
```
WDYT @lizzthabet @landism ?

Comment (lizzthabet) [reply desktop/release-notes.md:52]: thanks for calling this out, @trungutt! i might be more slightly more specific:

```
- Fixed the blank "An error occurred" container list on the Container Overview screen 
```

File: desktop/release-notes.md [hunk 5]
```diff
@@ -24,13 +24,94 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+
+- Fixed `docker build --quiet` not outputting the image identifier with the `containerd` integration.
+- Fixed image inspect not showing image labels with the `containerd` integration.
+- Increased the contrast between running and stopped container icons to make it easier for colourblind people to scan the containers list.
```
Comment (stephen-turner) [inline desktop/release-notes.md:56]: "color" not "colour" (we use American English).

File: desktop/release-notes.md [hunk 6]
```diff
@@ -24,13 +24,94 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+{% include release-date.html date="2022-01-12" %}
+
+> Download Docker Desktop
+>
+> {% include desktop-install.html %}
+
+### New
+
+- Extensions have moved from Beta to GA
+- Quick Search has moved from experimental to GA
```
Comment (trungutt) [inline desktop/release-notes.md:38]: I'd suggest:
```
New local images view has moved from experimental to GA
```
WDYT @stephanierifai ?

Comment (stephanierifai) [reply desktop/release-notes.md:38]: confirmed, let's add it!

File: desktop/release-notes.md (hunk 1 above)
Comment (aiordache) [reply desktop/release-notes.md:73]: @djs55 Sounds good to me.

File: desktop/release-notes.md [hunk 7]
```diff
@@ -24,13 +24,94 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+### Upgrades
+
+- [Compose v2.15.1](https://github.com/docker/compose/releases/tag/v2.15.1)
+- [Containerd v1.6.14](https://github.com/containerd/containerd/releases/tag/v1.6.14)
+- [Docker Engine v20.10.22](https://docs.docker.com/engine/release-notes/#201022)
+- [Docker Scan v0.23.0](https://github.com/docker/scan-cli-plugin/releases/tag/v0.23.0)
+- [Go 1.19.4](https://github.com/golang/go/releases/tag/go1.19.4)
+
+### Bug fixes and enhancements
+
```
Comment (gabriellavengeo) [inline desktop/release-notes.md:51]: Forgot to add notes to https://github.com/docker/pinata/pull/19937 - maybe something like `AF_VSOCK syscalls from within containers get blocked` in a separate security section. 

It's actually related to a vuln report (https://docker.atlassian.net/browse/POS-1175), basically same `AF_VSOCK` issue with accessing the host filesystem without `-v` that we had before but for WSL2 (where we can't do the one-connection hack)

File: desktop/release-notes.md (hunk 1 above)
Comment (mat007) [reply desktop/release-notes.md:73]: @djs55 do you have a link to an internal ticket or PR?

Comment (mat007) [reply desktop/release-notes.md:73]: Is there an internal ticket and/or a PR?

File: desktop/release-notes.md (hunk 2 above)
Comment (mat007) [reply desktop/release-notes.md:79]: @elapuyade?

File: desktop/release-notes.md (hunk 7 above)
Comment (mat007) [reply desktop/release-notes.md:51]: No related CVE?

Comment (gabriellavengeo) [reply desktop/release-notes.md:58]: No CVE yet as there’s still the problem with socketcall on 32-bit as seccomp doesn’t dereference pointer args. Discussing on #desktop-security-private how to go about it

File: desktop/release-notes.md (hunk 1 above)
Comment (MihaelaStoica) [reply desktop/release-notes.md:77]: POS-1356 I believe

Comment (mat007) [reply desktop/release-notes.md:77]: Ah, thanks, none of the linked PR had a release note then.

File: desktop/release-notes.md (hunk 2 above)
Comment (MihaelaStoica) [reply desktop/release-notes.md:83]: I don't think this is included in 4.16

File: desktop/release-notes.md [hunk 8]
```diff
@@ -24,13 +24,98 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+> {% include desktop-install.html %}
+
+### New
+
+- Extensions have moved from Beta to GA
+- Quick Search has moved from experimental to GA
+- Extensions are now included in Quick Search
+- Analyzing large images is now up to 4x faster
+- New local images view has moved from experimental to GA
+
```
Comment (djs55) [inline desktop/release-notes.md:42]:
```suggestion
- Add Rosetta for Linux as a Beta feature on MacOS 13+ for faster emulation of Intel-based images on Apple Silicon 
```

Comment (MihaelaStoica) [reply desktop/release-notes.md:42]: Do we need to say `MacOS 13+`? There is no macOS 14 yet, so probably `MacOS 13` is accurate at this point

File: desktop/release-notes.md (hunk 1 above)
Comment (mat007) [reply desktop/release-notes.md:77]: Thanks. So no for-mac issue related to any of these 3 new entries?

File: desktop/release-notes.md (hunk 8 above)
Comment (mat007) [reply desktop/release-notes.md:42]: Shorten a bit to
```
- Rosetta for Linux as a Beta feature on MacOS 13 for faster emulation of Intel-based images on Apple Silicon 
```
Is that all right?

Comment (djs55) [reply desktop/release-notes.md:43]: Suggestion updated

File: desktop/release-notes.md (hunk 1 above)
Comment (djs55) [reply desktop/release-notes.md:81]: I found one for the prompt but not for the other 2.

**Review by djs55 — APPROVED** [review]

File: desktop/release-notes.md [hunk 9]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+> {% include desktop-install.html %}
+
+### New
+
+- Extensions have moved from Beta to GA
+- Quick Search has moved from experimental to GA
+- Extensions are now included in Quick Search
+- Analyzing large images is now up to 4x faster
+- New local images view has moved from experimental to GA
+- Rosetta for Linux as a Beta feature on MacOS 13 for faster emulation of Intel-based images on Apple Silicon
```
Comment (aevesdocker) [inline desktop/release-notes.md:42]:
```suggestion
- New Beta feature for MacOS 13, Rosetta for Linux, has been added for faster emulation of Intel-based images on Apple Silicon.
```

File: desktop/release-notes.md [hunk 10]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+
+#### For all platforms
+
+- Fixed `docker build --quiet` not outputting the image identifier with the `containerd` integration.
+- Fixed image inspect not showing image labels with the `containerd` integration.
+- Increased the contrast between running and stopped container icons to make it easier for colorblind people to scan the containers list.
+- Fixed a bug where the user is prompted for new HTTP proxy credentials repeatedly until Docker Desktop is restarted.
+- Added a diagnostics command `com.docker.diagnose login` to check HTTP proxy configuration.
+- Fixed actions on compose stack not working properly. Fixes [docker/for-mac#6566](https://github.com/docker/for-mac/issues/6566).
+- Fixed the Docker Desktop dashboard trying at startup to get disk usage information and display an error banner before the engine was running.
```
Comment (aevesdocker) [inline desktop/release-notes.md:63]:
```suggestion
- Fixed the Docker dashboard trying at startup to get disk usage information and display an error banner before the engine was running.
```

**Review by trungutt — DISMISSED** [review]

File: desktop/release-notes.md [hunk 11]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Added an informational banner with instructions to opt-out of experimental feature access next to all experimental features.
```
Comment (aevesdocker) [inline desktop/release-notes.md:64]:
```suggestion
- Added an informational banner with instructions on how to opt-out of experimental feature access next to all experimental features.
```

File: desktop/release-notes.md [hunk 12]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Support downloading Kubernetes images via an HTTP proxy.
```
Comment (aevesdocker) [inline desktop/release-notes.md:65]:
```suggestion
- Docker Desktop now supports downloading Kubernetes images via an HTTP proxy.
```

File: desktop/release-notes.md [hunk 13]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Fixed tooltips to not block action buttons. Fixes [docker/for-mac#6516](https://github.com/docker/for-mac/issues/6516).
+- Fixed the blank "An error occurred" container list on the Container Overview screen
```
Comment (aevesdocker) [inline desktop/release-notes.md:67]:
```suggestion
- Fixed the blank "An error occurred" container list on the **Container** view.
```

File: desktop/release-notes.md [hunk 14]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+
+#### For Mac
+
+- Fixed the Docker engine not starting when Enhanced Container Isolation is enabled if the legacy `osxfs` implementation is used for file sharing.
+- Fixed files created on VirtioFS having the executable bit set. Fixes [docker/for-mac#6614](https://github.com/docker/for-win/issues/6614).
+- Added back a way to uninstall Docker Desktop from the command line. Fixes [docker/for-mac#6598](https://github.com/docker/for-mac/issues/6598).
+- Fixed hardcoded `/usr/bin/kill`. Fixes [docker/for-mac#6589](https://github.com/docker/for-mac/issues/6589).
+- Fixed truncation (for example with the `truncate` command) of very large files (> 38GB) shared on VirtioFS with an incorrect size.
+- Changed the disk image size in the Settings to use the decimal system (base 10) to coincide with how Finder displays disk capacity.
```
Comment (aevesdocker) [inline desktop/release-notes.md:76]:
```suggestion
- Changed the disk image size in **Settings** to use the decimal system (base 10) to coincide with how Finder displays disk capacity.
```

File: desktop/release-notes.md [hunk 15]
```diff
@@ -24,13 +24,102 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
...
+- Fixed Docker crash under network load. Fixes [docker/for-mac#6530](https://github.com/docker/for-mac/issues/6530).
+- Fixed an issue causing Docker to prompt the user to install the `/var/run/docker.sock` symlink after every reboot.
+- Ensured the Login Item which installs the `/var/run/docker.sock` symlink is signed.
+- Fixed bug where `$HOME/.docker` was removed on factory reset.
+
+### For Windows
+
+- Fixed `docker build` hanging while printing "load metadata for". Fixes [docker/for-win#10247](https://github.com/docker/for-win/issues/10247).
+- Fixed typo in diagnose.exe output (https://github.com/docker/for-win/issues/13107).
```
Comment (aevesdocker) [inline desktop/release-notes.md:85]:
```suggestion
- Fixed typo in diagnose.exe output Fixes [docker/for-win#13107](https://github.com/docker/for-win/issues/13107).
```

**Review by aevesdocker — COMMENTED** [review]
Just a few nits

**Review by aevesdocker — APPROVED** [review]

**Pull request merged**